test/sw_mgmt_spec.rb

```diff
@@ -225,30 +226,44 @@
     end
 
     it "copies old NCC credentials at upgrade" do
-      expect(File).to receive(:exist?).with(File.join(root_dir, target_dir, "NCCcredentials"))
-        .and_return(true)
-      expect(File).to receive(:exist?).with(File.join(root_dir, target_dir, "SCCcredentials"))
-        .and_return(false)
+      expect(Dir).to receive(:[]).with(File.join(root_dir, target_dir, "*"))
+        .and_return([ncc_credentials])
+      expect(File).to receive(:exist?).with(ncc_credentials).and_return(true)
 
-      expect(subject).to receive(:`).with("cp -a " + File.join(root_dir, target_dir,
-        "NCCcredentials") + " " + File.join(target_dir, "SCCcredentials"))
+      expect(subject).to receive(:`).with("cp -a " + ncc_credentials + " " +
+        File.join(target_dir, "SCCcredentials"))
       expect(SUSE::Connect::YaST).to receive(:credentials).and_return(OpenStruct.new)
 
       expect { subject.copy_old_credentials(root_dir) }.to_not raise_error
     end
 
     it "copies old SCC credentials at upgrade" do
-      expect(File).to receive(:exist?).with(File.join(root_dir, target_dir, "NCCcredentials"))
-        .and_return(false)
-      expect(File).to receive(:exist?).with(File.join(root_dir, target_dir, "SCCcredentials"))
-        .and_return(true)
-
-      expect(subject).to receive(:`).with("cp -a " + File.join(root_dir, target_dir,
-        "SCCcredentials") + " " + File.join(target_dir, "SCCcredentials"))
+      scc_credentials = File.join(root_dir, target_dir, "SCCcredentials")
+      expect(Dir).to receive(:[]).with(File.join(root_dir, target_dir, "*"))
+        .and_return([scc_credentials])
+      expect(File).to receive(:exist?).with(scc_credentials).and_return(true)
+      expect(File).to receive(:exist?).with(ncc_credentials).and_return(false)
+
+      expect(subject).to receive(:`).with("cp -a " + scc_credentials + " " +
+        File.join(target_dir, "SCCcredentials"))
       expect(SUSE::Connect::YaST).to receive(:credentials).and_return(OpenStruct.new)
 
       expect { subject.copy_old_credentials(root_dir) }.to_not raise_error
     end
+
+    it "copies old SCC credentials at upgrade" do
+      smt_credentials = File.join(root_dir, target_dir, "SMT-http_smt_example_com")
+      expect(Dir).to receive(:[]).with(File.join(root_dir, target_dir, "*"))
+        .and_return([smt_credentials])
+      expect(File).to receive(:exist?).with(smt_credentials).and_return(true)
+      expect(File).to receive(:exist?).with(ncc_credentials).and_return(false)
+
+      expect(subject).to receive(:`).with("cp -a " + smt_credentials + " " +
+        File.join(target_dir, "SMT-http_smt_example_com"))
+      expect(SUSE::Connect::YaST).to receive(:credentials).and_return(OpenStruct.new)
+
+      subject.copy_old_credentials(root_dir)
+    end
```

Comment:
and if you want to be so explicit, add also test for RMT.

Reply:
Um, that's not necessary, both RMT and SMT use dynamic name of the credentials (usually the host name). So it's enough to test anything different to the standard `SCCcredentials`.